uaclient/cli.py

```diff
@@ -1372,22 +1377,18 @@ def action_api(args, *, cfg):
 
 
 @assert_root
-@assert_lock_file("pro auto-attach")
-def action_auto_attach(args, *, cfg):
-    if cfg.is_attached:
-        raise exceptions.AlreadyAttachedOnPROError()
-
-    skip_auto_attach = actions.should_disable_auto_attach(cfg)
-    if skip_auto_attach:
-        return 0
-
-    instance = actions.get_cloud_instance(cfg)
-
+def action_auto_attach(args, *, cfg: config.UAConfig) -> int:
     try:
-        actions.auto_attach(cfg, instance)
+        _full_auto_attach(
+            FullAutoAttachOptions(),
+            cfg=cfg,
+            mode=event_logger.EventLoggerMode.CLI,
+        )
     except exceptions.UrlError:
         event.info(messages.ATTACH_FAILURE.msg)
         return 1
+    except api_exceptions.AutoAttachDisabledError:
+        return 0
```

Comment:
I think we can return 1 for this error as well, so I think we can just let the main_error_handler handle this as well - which means we don't need these lines :D
```suggestion
```